src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  function Clone(copy) {
+    return Object.assign({}, copy);
+  }
+
+  const actionSteps = [];
+  let copyState = Object.assign({}, state);
+
+  for (const data of actions) {
+    switch (data.type) {
+      case 'clear':
+        copyState = {};
+
+        actionSteps.push(Clone(copyState));
+        break;
+
+      case 'removeProperties':
+        for (const remove of data.keysToRemove) {
+          delete copyState[remove];
+        }
+
+        actionSteps.push(Clone(copyState));
+        break;
+
+      case 'addProperties':
+        Object.assign(copyState, data.extraData);
+
+        actionSteps.push(Clone(copyState));
+    }
+  }
+
```

Comment:
```suggestion
        break;
        
      case 'removeProperties':
        for (const remove of data.keysToRemove) {
          delete copyState[remove];
        }
        break;

      case 'addProperties':
        Object.assign(copyState, data.extraData);
    }
            actionSteps.push(Clone(copyState));
  }

```

move the same logic outside, for one iteration it will be only one case, so it will work every time, no need to repeat yourself

and add default case : break